src/m365/file/commands/file-copy.ts

```diff
@@ -0,0 +1,200 @@
+  webUrl: string;
+  sourceUrl: string;
+  targetUrl: string;
+  newName?: string;
+  nameConflictBehavior?: string;
+}
+
+class FileCopyCommand extends GraphCommand {
+  private readonly nameConflictBehaviorOptions = ['fail', 'replace', 'rename'];
+
+  public get name(): string {
+    return commands.COPY;
+  }
+
+  public get description(): string {
+    return 'Copies a file to another location using the Microsoft Graph';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        webUrl: typeof args.options.webUrl !== 'undefined',
+        sourceUrl: typeof args.options.sourceUrl !== 'undefined',
+        targetUrl: typeof args.options.targetUrl !== 'undefined',
+        newName: typeof args.options.newName !== 'undefined',
+        nameConflictBehavior: typeof args.options.nameConflictBehavior !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-u, --webUrl <webUrl>' },
+      { option: '-s, --sourceUrl <sourceUrl>' },
+      { option: '-t, --targetUrl <targetUrl>' },
+      { option: '--newName [newName]' },
+      { option: '--nameConflictBehavior [nameConflictBehavior]', autocomplete: this.nameConflictBehaviorOptions }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.nameConflictBehavior && this.nameConflictBehaviorOptions.indexOf(args.options.nameConflictBehavior) === -1) {
+          return `${args.options.nameConflictBehavior} is not a valid nameConflictBehavior value. Allowed values: ${this.nameConflictBehaviorOptions.join(', ')}.`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const { webUrl, sourceUrl, targetUrl, newName, verbose } = args.options;
+
+      const sourcePath: string = this.getAbsoluteUrl(webUrl, sourceUrl);
+      const destinationPath: string = this.getAbsoluteUrl(webUrl, targetUrl);
+
+      if (this.verbose) {
+        logger.logToStderr(`Copying file '${sourcePath}' to '${destinationPath}'...`);
+      }
+
+      const copyUrl: string = await this.getCopyUrl(args.options, sourcePath, logger);
+      const { targetDriveId, targetItemId } = await this.getTargetDriveAndItemId(webUrl, targetUrl, logger, verbose);
+
+      const requestOptions: CliRequestOptions = {
+        url: copyUrl,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          parentReference: {
+            driveId: targetDriveId,
+            id: targetItemId
+          }
+        }
+      };
+
+      if (args.options.newName) {
```

Comment:
I was playing around with the command and I noticed that how it is implemented now is that we also need to add the file extension in the new name 🤔.
So for example in library A I have a file test.docx and now when I used this command to copy the file to library B specifying the newName as:
- `testB.docx` all worked fine
- `testB` I got a new not defined file in library B
- `testB.pdf` I got a not working pdf in library B

I was thinking that if it is not possible to change the format anyway with this endpoint does it make sense to specify it 🤔. I mean if I had a `docx` the new file will need to be a `docx` anyway so would it be better if in the `newName` option we just specify the name itself and not the name and the file type? 
@Saurabh7019 I am really interested in what you think and what you feel better for the user 🙂?

Reply:
It makes sense to simplify the newName option. I have now adjusted the code to handle the extension based on the source file.